src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -317,3 +324,79 @@ bool GlobalAppSettings::ShouldUsePersistedLayout() const
 {
     return FirstWindowPreference() == FirstWindowPreference::PersistedWindowLayout && !IsolatedMode();
 }
+
+void GlobalAppSettings::_logSettingSet(std::string_view setting)
+{
+    if (setting == "theme")
+    {
+        if (_Theme.has_value())
+        {
+            // ThemePair always has a Dark/Light value,
+            // so we need to check if they were explicitly set
+            if (_Theme->DarkName() == _Theme->LightName())
+            {
+                _changeLog.insert(setting);
+            }
+            else
+            {
+                _changeLog.insert(std::string_view{ fmt::format(FMT_COMPILE("{}.{}"), setting, "dark") });
+                _changeLog.insert(std::string_view{ fmt::format(FMT_COMPILE("{}.{}"), setting, "light") });
+            }
+        }
+    }
+    else if (setting == "newTabMenu")
+    {
+        if (_NewTabMenu.has_value())
+        {
+            for (const auto& entry : *_NewTabMenu)
+            {
+                std::string entryType;
+                switch (entry.Type())
+                {
+                case NewTabMenuEntryType::Profile:
+                    entryType = "profile";
+                    break;
+                case NewTabMenuEntryType::Separator:
+                    entryType = "separator";
+                    break;
+                case NewTabMenuEntryType::Folder:
+                    entryType = "folder";
+                    break;
+                case NewTabMenuEntryType::RemainingProfiles:
+                    entryType = "remainingProfiles";
+                    break;
+                case NewTabMenuEntryType::MatchProfiles:
+                    entryType = "matchProfiles";
+                    break;
+                case NewTabMenuEntryType::Action:
+                    entryType = "action";
+                    break;
+                case NewTabMenuEntryType::Invalid:
+                    // ignore invalid
+                    continue;
+                }
+                _changeLog.insert(std::string_view{ fmt::format(FMT_COMPILE("{}.{}"), setting, entryType) });
+            }
+        }
+    }
+    else
+    {
+        _changeLog.insert(setting);
+    }
+}
+
+void GlobalAppSettings::_logSettingIfSet(std::string_view setting, const bool isSet)
+{
+    if (isSet)
+    {
+        _logSettingSet(setting);
+    }
+}
+
+void GlobalAppSettings::LogSettingChanges(std::set<std::string_view>& changes, std::string_view& context) const
+{
+    for (const auto& setting : _changeLog)
+    {
+        changes.emplace(fmt::format(FMT_COMPILE("{}.{}"), context, setting));

```

Comment:
The data is getting corrupted into `ÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝÝ` when we go through the fmt::format call. I suspect that I'm using it wrong, but I can't figure out why 🫤

Reply:
Ý is U+00DD, which means your entire string is filled with 0xDD bytes. 0xDD is the "freed memory" marker by the debug mode allocator.

Glancing through the file I noticed that you're `fmt::format`ing strings and then insert them as string-views into the vector. That can't work, because string-views don't hold onto the container/owner of the string and so the string will get dropped at the end of the line. My suggestion to use string-views was in the case that we only log the settings keys. I hadn't considered that we may need to log nested keys... 😓

There's some other place that uses owned strings to store literals. My recommendation is to use `string_view` when referring to a set of string literals (and only those), or when you want to locally process a string, and you know for a fact you're holding onto the owned string as well. Otherwise, I'd always use an owned string (be it `string` or `hstring`, etc.)